File: src/main/java/org/noxet/noxetserver/commands/realms/smp/SMP.java
package org.noxet.noxetserver.commands.realms.smp;

import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.noxet.noxetserver.RealmManager;
import org.noxet.noxetserver.messaging.ErrorMessage;

import static org.noxet.noxetserver.RealmManager.migrateToRealm;

@SuppressWarnings("ALL")
public class SMP implements CommandExecutor {
    @Override
    public boolean onCommand(CommandSender commandSender, Command command, String s, String[] strings) {
        if(!(commandSender instanceof Player)) {
            new ErrorMessage(ErrorMessage.ErrorType.COMMON, "Only players can be sent to the SMP server.").send(commandSender);
            return true;
        }

        migrateToRealm((Player) commandSender, RealmManager.Realm.SMP);

        return true;
    }
}

File: src/main/java/org/noxet/noxetserver/commands/realms/smp/SMP.scala
package org.noxet.noxetserver.commands.realms.smp

import org.bukkit.command.{Command, CommandExecutor, CommandSender}
import org.bukkit.entity.Player
import org.noxet.noxetserver.messaging.ErrorMessage
import org.noxet.noxetserver.realm.RealmManager
import org.noxet.noxetserver.realm.RealmManager.migrateToRealm

class SMP extends CommandExecutor:
  override def onCommand(commandSender: CommandSender, command: Command, s: String, strings: Array[String]): Boolean =
    commandSender match
      case p: Player => migrateToRealm(p, RealmManager.Realm.SMP)
      case _ => ErrorMessage(ErrorMessage.ErrorType.COMMON, "Only players can be sent to the SMP server.").send(commandSender)
    true